### DISCUSSION TITLE
Connect to host via cloudflared tunnel?

### DISCUSSION THREAD

**Discussion by decky-grande** [cockpit-project/cockpit#21031] [General]
I could access the cockpit via https://192.168.7.123:5030

I want to create a tunnel with ZeroTrust to `cockpit.domain.ext` But when I point the tunnel to `https://192.168.7.123:5030`, it would say `Bad gateway Error code 502`, and if I target `http://192.168.7.123:5030` it would say `always redirected` error

I tried to puta config it to  /etc/cockpit/cockpit.conf
using 

```[WebService]
Origins = https://coccpit.domain.ext wss://coccpit.domain.ext   
ProtocolHeader = X-Forwarded-Proto
AllowUnencrypted = false
```

But it causes the server can't even be opened from the local network

Any guidance to make it work with Cloudflare zero trust tunnel?

**Comment by martinpitt** [comment]
I can't help you with cloudflare (no experience at all). But your current configuration says that you *only* want to access Cockpit via `coccpit.domain.ext` [sic]. If you *also* want to access it locally, add `https://192.168.7.123:5030` and wss://... to `Origins =`.

**Comment by vitalivu992** [comment]
Please make sure you configure the route in the tunnel with
- `https`
- `No TLS Verify`. 

and don't set value for:
- Origin Server Name (null by default)
- HTTP Host Header (null by default)
